SpatialGDK/Source/SpatialGDKEditor/Public/SpatialGDKEditorSettings.h

```diff
@@ -7,6 +7,190 @@
+	}
+
+	/** The size of the simulation in meters for the auto generated launch config. */
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Dimensions"))
+	FIntPoint Dimensions;
+
+	/** The size of the grid squares that the world is divided into, in “world units” (an arbitrary unit that workers can interpret as they choose. */
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Chunk edge length in meters"))
+	int32 ChunkEdgeLengthMeters;
+
+	/** The time between streaming query updates (in seconds). */
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Streaming query interval"))
+	int32 StreamingQueryInterval;
+
+	/** The frequency in seconds to write snapshots of the simulated world. */
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Snapshot write period in seconds"))
+	int32 SnapshotWritePeriodSeconds;
+
+	/** Legacy non-worker flag configurations. */
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Legacy flags"))
+	TMap<FString, FString> LegacyFlags;
+
+	/** Legacy JVM configurations. */
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "Legacy java parameters"))
+	TMap<FString, FString> LegacyJavaParams;
+};
+
+USTRUCT()
+struct FWorkerPermissionsSection
+{
+	GENERATED_BODY()
+
+	FWorkerPermissionsSection()
+		: bAllPermissions(true)
+		, bAllowEntityCreation(true)
+		, bAllowEntityDeletion(true)
+		, bAllowEntityQuery(true)
+		, Components()
+	{
+
+	}
+
+	/** Gives all permissions to a worker. */
+	UPROPERTY(EditAnywhere, config, meta = (ConfigRestartRequired = false, DisplayName = "All"))
+	bool bAllPermissions;
+
+	/** Enables a worker to create new entities */
+	UPROPERTY(EditAnywhere, config, meta = (EditCondition = "!bAllPermissions", ConfigRestartRequired = false, DisplayName = "Allow entity creation"))
+	bool bAllowEntityCreation;
+
+	/** Enables a worker to delete entities */
+	UPROPERTY(EditAnywhere, config, meta = (EditCondition = "!bAllPermissions", ConfigRestartRequired = false, DisplayName = "Allow entity deletion"))
+	bool bAllowEntityDeletion;
+
+	/** Controls what components can be returned from entity queries performed by the worker. If an entity query specifies other components to be returned then the query will fail. */
+	UPROPERTY(EditAnywhere, config, meta = (EditCondition = "!bAllPermissions", ConfigRestartRequired = false, DisplayName = "Allow entity query"))
+	bool bAllowEntityQuery;
+
+	/** Specifies which components can be returned in the query result. */
+	UPROPERTY(EditAnywhere, config, meta = (EditCondition = "!bAllPermissions", ConfigRestartRequired = false, DisplayName = "Component queries"))
+	TArray<FString> Components;
+};
+
+USTRUCT()
+struct FLoginRateLimitSection
+{
+	GENERATED_BODY()
+
+	FLoginRateLimitSection()
+		: Duration()
+		, RequestsPerDuration(0)
+	{
+
```

Comment:
Ah this must be a deliberate newline - is this consistent with what we do elsewhere in the GDK?